File: tests/fixtures/gold/r/public_surface.R
# Package: geometry

Point <- R6Class("Point",
  public = list(
    x = 0,
    y = 0
  )
)

distance <- function(a, b) {
  sqrt((a$x - b$x)^2 + (a$y - b$y)^2)
}

area = function(shape, ...) {
  0
}

setGeneric("describe", function(obj) standardGeneric("describe"))

.helper <- function(x) {
  x * 2
}

File: tests/fixtures/gold/julia/public_surface.jl
module Geometry

const VERSION = "1.0"

abstract type Shape end

struct Point
    x::Float64
    y::Float64
end

function distance(a::Point, b::Point)
    return sqrt((a.x - b.x)^2 + (a.y - b.y)^2)
end

area(p::Point) = 0.0

macro logged(expr)
    return expr
end

end
